### PR TITLE
Search functions

### LINKED ISSUE #10: create search queries

### PR DESCRIPTION
**Pull request by lnmylos**
Closes #10

### CODE REVIEW

File: models/token_data.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class ReadConfig(BaseModel):
     """the ReadConfig class"""
 
-    config: dict[str, str] = load_config(Path("config/token.json"))
+    config: dict[str, str] = load_config(Path("config/token.json"))  # type: ignore
```
Comment (Goldensteev) [inline models/token_data.py:13]: linters complaining about typing for this line?

Comment (lnmylos) [reply models/token_data.py:13]: yes

Comment (Goldensteev) [reply models/token_data.py:13]: weird on my machine linter not complaining? what's the message?

File: models/token_data.py [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
-from utils import load_config
+from utils import load_json_file
 
 
 class ReadConfig(BaseModel):
-    """the ReadConfig class"""
+    """the ReadConfig Data class"""
 
-    config: dict[str, str] = load_config(Path("config/token.json"))
+    config: dict[str, str] = load_json_file(Path("config", "token.json"))
```
Comment (Goldensteev) [inline models/token_data.py:13]: testing agnostic file writing paths using `pathlib.Path()`

File: utils/fhir_api.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    Returns:
+        url (str): the searching query
+    """
+
+    with open("../config/settings.yml", "r", encoding="utf-8") as yaml_file:
+        config = yaml.safe_load(yaml_file)
+
+    url_base = config["fhir_api_base"]
+    url_endpoint = config["endpoints"].get(endpoint)
```
Comment (Goldensteev) [inline utils/fhir_api.py:23]: I feel using `.get()` think we could benefit with most `dict` key calls make things more robust perhaps?
even raising an error in the case where the URLs are not correctly loaded?

**Comment by lnmylos** [comment]
@Goldensteev Ready for revision :)

File: utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        if i["name"] == "code":
+            code = i["valueCode"]
+
+        if i["name"] == "display":
+            label = i["valueString"]
+
+        if i["name"] == "name":
+            vocabulary = i["valueString"]
+
+    return {"code": code, "label": label, "vocabulary": vocabulary}
```
Comment (Goldensteev) [inline utils/utils.py:48]: i would have written something like this :

```python 
mapping = {
    "code": "valueCode",
    "display": "valueString",
    "name": "valueString",
}

def map_parameters(result, mapping):
    mapped_result = {key: None for key in mapping.values()}

    for param in result.get("parameter", []):
        name = param.get("name")
        mapped_key = mapping.get(name)
        if mapped_key:
            mapped_result[mapped_key] = param.get(mapped_key)

    return mapped_result
```

File: utils/utils.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                child = parameter["part"][1]["valueCode"]
+                children.append(child)
+
+    for parent in parents:
+        codes_parents = search_code(token=token, url=url, code=parent)  # type: ignore
+        relations["parents"].append(get_code_label(codes_parents))  # type: ignore
+
+    for child in children:
+        codes_children = search_code(token=token, url=url, code=child)  # type: ignore
+        relations["children"].append(get_code_label(codes_children))  # type: ignore
```
Comment (Goldensteev) [inline utils/utils.py:87]: this snippet could be changed to have the codes be batched together

```python
codes = {code: [] for code in set(parents + children)}
for code in codes.keys():
    code_result = search_code(token=token, url=url, code=code)
    codes[code] = get_code_label(code_result)
relations["parents"] = [codes[parent] for parent in parents]
relations["children"] = [codes[child] for child in children]
```

File: utils/utils.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+        if parameter["name"] == "property":
+
+            if parameter["part"][0]["valueCode"] == "parent":
+                parent = parameter["part"][1]["valueCode"]
+                parents.append(parent)
+
+            if parameter["part"][0]["valueCode"] == "child":
+                child = parameter["part"][1]["valueCode"]
+                children.append(child)
```
Comment (Goldensteev) [inline utils/utils.py:79]: all those nested ifs are burning my eyes haha

Comment (lnmylos) [reply utils/utils.py:79]: check standard fhir resource "parameters":  https://hl7.org/fhir/r4/parameters.html

File: utils/fhir_api.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    Args:
+        config_path (str): Defaults to PATH_CONFIG.
+
+    Returns:
+        str: dictionary with the urls and endpoints.
+    """
+
+    with open(config_file_path, "r", encoding="utf-8") as yaml_file:
+        config_data = yaml.safe_load(yaml_file)
+    return config_data
```
[the remark this answers is not in the document]
Comment (lnmylos) [reply utils/fhir_api.py:24]: This function will probably disappear with once the front is ready. 

This is only used to work with the functions in a python env without the front.   :)

**Pull request merged**